Mathlib/CategoryTheory/Sites/Sheaf.lean

```diff
@@ -77,6 +77,13 @@ def IsSheaf (P : Cᵒᵖ ⥤ A) : Prop :=
   ∀ E : A, Presieve.IsSheaf J (P ⋙ coyoneda.obj (op E))
 #align category_theory.presheaf.is_sheaf CategoryTheory.Presheaf.IsSheaf
 
+attribute [local instance] ConcreteCategory.hasCoeToSort ConcreteCategory.instFunLike in
+/-- The condition that a presheaf with values in a concrete category is separated for
+a Grothendieck topology. -/
+def IsSeparated (P : Cᵒᵖ ⥤ A) [ConcreteCategory A] : Prop :=
+  ∀ (X : C) (S : Sieve X) (_ : S ∈ J X) (x y : P.obj (op X)),
+    (∀ (Y : C) (f : Y ⟶ X) (_ : S f), P.map f.op x = P.map f.op y) → x = y
+
```

Comment:
Shouldn't this be defined similarly as `CategoryTheory.Presheaf.IsSheaf` for arbitrary categories and then show that they are equivalent for concrete categories?

Reply:
I had considered doing so. It could express that the restriction maps given by a covering sieve are monomorphic families. I think there would be little use for this notion, and I do not know how to relate this categorical notion and the concrete one. (If it were only about a single monomorphism, it would suffice to assume that the forget functor preserves mono (or finite limits), but in general, as we do not have reasonable bounds about the cardinality/universe of covering families, I do not think it is worth introducing this notion.)

Reply:
I see. How about `Presieve.IsSeparated J (P ⋙ forget A)` then?

Reply:
Thanks for the suggestion!